src/main/interface/msp.c

```diff
@@ -799,93 +704,79 @@ bool mspProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst)
+        // write flightModeFlags header. Lowest 4 bits contain number of bytes that follow
+        // header is emited even when all bits fit into 32 bits to allow future extension
+        int byteCount = (flagBits - 32 + 7) / 8;        // 32 already stored, round up
+        byteCount = constrain(byteCount, 0, 15);        // limit to 16 bytes (128 bits)
+        sbufWriteU8(dst, byteCount);
+        sbufWriteData(dst, ((uint8_t*)&flightModeFlags) + 4, byteCount);
+        // Write arming disable flags
+        // 1 byte, flag count
+        sbufWriteU8(dst, ARMING_DISABLE_FLAGS_COUNT);
+        // 4 bytes, flags
+        const uint32_t armingDisableFlags = getArmingDisableFlags();
+        sbufWriteU32(dst, armingDisableFlags);
+    }
+    break;
+    case MSP_RAW_IMU: {
+        // Hack scale due to choice of units for sensor data in multiwii
+        uint8_t scale = 1;
 #ifndef USE_GYRO_IMUF9001
-            if (acc.dev.acc_1G > 512*4) {
-                scale = 8;
-            } else if (acc.dev.acc_1G > 512*2) {
-                scale = 4;
-            } else if (acc.dev.acc_1G >= 512) {
-                scale = 2;
-            }
+        if (acc.dev.acc_1G > 512 * 4) {
+            scale = 8;
+        } else if (acc.dev.acc_1G > 512 * 2) {
+            scale = 4;
+        } else if (acc.dev.acc_1G >= 512) {
+            scale = 2;
+        }
 #endif //USE_GYRO_IMUF9001
-
-            for (int i = 0; i < 3; i++) {
-                sbufWriteU16(dst, lrintf(acc.accADC[i] / scale));
-            }
-            for (int i = 0; i < 3; i++) {
-                sbufWriteU16(dst, gyroRateDps(i));
-            }
-            for (int i = 0; i < 3; i++) {
-                sbufWriteU16(dst, lrintf(mag.magADC[i]));
-            }
+        for (int i = 0; i < 3; i++) {
+            sbufWriteU16(dst, lrintf(acc.accADC[i] / scale));
         }
-        break;
-
-    case MSP_NAME:
-        {
-            // Show warning for DJI OSD instead of pilot name
-            // works if osd warnings enabled, osd_warn_dji is on and usb is not connected
-            if (osdWarnDjiEnabled()) {
-                sbufWriteString(dst, djiWarningBuffer);
-                break;
-            }
-
-            const int nameLen = strlen(pilotConfig()->name);
-            for (int i = 0; i < nameLen; i++) {
-                sbufWriteU8(dst, pilotConfig()->name[i]);
-            }
+        for (int i = 0; i < 3; i++) {
+            sbufWriteU16(dst, gyroRateDps(i));
         }
-        break;
-
+        for (int i = 0; i < 3; i++) {
+            sbufWriteU16(dst, lrintf(mag.magADC[i]));
+        }
+    }
+    break;
+    case MSP_NAME: {
+        // Show warning for DJI OSD instead of pilot name
+        // works if osd warnings enabled, osd_warn_dji is on and usb is not connected
+        if (osdWarnDjiEnabled()) {
+            sbufWriteString(dst, djiWarningBuffer);
+            break;
+        }
+        const int nameLen = strlen(pilotConfig()->name);
```

Comment:
Codacy found an issue: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=6326343)